src/marqo/s2_inference/clip_utils.py

```diff
@@ -156,15 +185,195 @@ def __init__(self, model_type: str = "ViT-B/32", device: str = 'cpu',  embedding
+            self.tokenizer = clip.tokenize
+
+
+        except RuntimeError:
+            logger.info("Can not load clip model. Try custom clip model loading.")
+
+            path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+            if path is None:
+                raise InvalidModelPropertiesError("Model path `url` or `localpath` is not provided. Can not load generic model.")
+            elif os.path.isfile(path):
+                self.model_path = path
+            elif validators.url(path):
+                self.model_path = download_pretrained_from_url(path)
+
+            self.jit = self.model_properties.get("jit", False)
+            self.device = self.model_properties.get("device", "cpu")
+            self.mean = self.model_properties.get("mean", None)
+            self.std = self.model_properties.get("std", None)
+
+
+            try:
+                self.model, self.preprocess = self.openai_clip_load()
+                self.model.eval()
+            except EOFError:
+                self.model, self.preprocess = self.open_clip_load()
+                self.model.eval()
+
+            self.tokenizer = self.load_tokenizer()
+
+
+    def openai_clip_load(self):
+        with open(self.model_path, 'rb') as opened_file:
+            try:
+                # loading JIT archive
+                model = torch.jit.load(opened_file, map_location=self.device if self.jit else "cpu").eval()
+                state_dict = None
+            except RuntimeError:
+                # loading saved state dict
+                if self.jit:
+                    logger.info(f"File {self.model_path} is not a JIT archive. Loading as a state dict instead")
+                    self.jit = False
+                state_dict = torch.load(opened_file, map_location="cpu")
+
+        if not self.jit:
+            model = build_model(state_dict or model.state_dict()).to(self.device)
+            if str(self.device) == "cpu":
+                model.float()
+            return model, _get_transform(model.visual.input_resolution, self.mean, self.std)
+
+        # patch the device names
+        device_holder = torch.jit.trace(lambda: torch.ones([]).to(torch.device(self.device)), example_inputs=[])
+        device_node = [n for n in device_holder.graph.findAllNodes("prim::Constant") if "Device" in repr(n)][-1]
+
+        def patch_device(module):
+            try:
+                graphs = [module.graph] if hasattr(module, "graph") else []
+            except RuntimeError:
+                graphs = []
+
+            if hasattr(module, "forward1"):
+                graphs.append(module.forward1.graph)
+
+            for graph in graphs:
+                for node in graph.findAllNodes("prim::Constant"):
+                    if "value" in node.attributeNames() and str(node["value"]).startswith("cuda"):
+                        node.copyAttributes(device_node)
+
+        model.apply(patch_device)
```

Comment:
is it possible to use the original load functions instead? https://github.com/openai/CLIP/blob/main/clip/clip.py#L94